ios/Classes/codec/AblyFlutterReader.m

```diff
@@ -54,87 +66,102 @@ use an explicitly received null from Dart (manifesting this side as a nil id)
+    READ_VALUE(o, authParams, jsonDict, TxClientOptions.authParams);
+    READ_VALUE(o, queryTime, jsonDict, TxClientOptions.queryTime);
 
     // ClientOptions
-    READ_VALUE(o, clientId);
-    ON_VALUE(^(const id value) { o.logLevel = _logLevel(value); });
+    READ_VALUE(o, clientId, jsonDict, TxClientOptions.clientId);
+    ON_VALUE(^(const id value) { o.logLevel = _logLevel(value); }, jsonDict, TxClientOptions.logLevel);
     //TODO log handler
-    READ_VALUE(o, tls);
-    READ_VALUE(o, restHost);
-    READ_VALUE(o, realtimeHost);
-    [self readValue]; // TODO READ_VALUE(o, port); // NSInteger
-    [self readValue]; // TODO READ_VALUE(o, tlsPort); // NSInteger
-    READ_BOOL(o, autoConnect);
-    READ_VALUE(o, useBinaryProtocol);
-    READ_VALUE(o, queueMessages);
-    READ_VALUE(o, echoMessages);
-    READ_VALUE(o, recover);
-    READ_VALUE(o, environment);
-    READ_VALUE(o, idempotentRestPublishing);
-    [self readValue]; // TODO READ_VALUE(o, httpOpenTimeout); // NSTimeInterval
-    [self readValue]; // TODO READ_VALUE(o, httpRequestTimeout); // NSTimeInterval
-    [self readValue]; // TODO READ_VALUE(o, httpMaxRetryCount); // NSUInteger
-    [self readValue]; // TODO READ_VALUE(o, realtimeRequestTimeout); // NSTimeInterval
-    READ_VALUE(o, fallbackHosts);
-    READ_VALUE(o, fallbackHostsUseDefault);
-    [self readValue]; // TODO READ_VALUE(o, fallbackRetryTimeout); // property not found
-    READ_VALUE(o, defaultTokenParams);
-    [self readValue]; // TODO READ_VALUE(o, channelRetryTimeout); // NSTimeInterval
-    [self readValue]; // TODO READ_VALUE(o, transportParams); // property not found
-    //    [self readValue]; // TODO READ_VALUE(o, asyncHttpThreadpoolSize); // property not found
-    //    READ_VALUE(o, pushFullWait);
+    READ_VALUE(o, tls, jsonDict, TxClientOptions.tls);
+    READ_VALUE(o, restHost, jsonDict, TxClientOptions.restHost);
+    READ_VALUE(o, realtimeHost, jsonDict, TxClientOptions.realtimeHost);
+    READ_BOOL(o, autoConnect, jsonDict, TxClientOptions.autoConnect);
+    READ_VALUE(o, useBinaryProtocol, jsonDict, TxClientOptions.useBinaryProtocol);
+    READ_VALUE(o, queueMessages, jsonDict, TxClientOptions.queueMessages);
+    READ_VALUE(o, echoMessages, jsonDict, TxClientOptions.echoMessages);
+    READ_VALUE(o, recover, jsonDict, TxClientOptions.recover);
+    READ_VALUE(o, environment, jsonDict, TxClientOptions.environment);
+    READ_VALUE(o, idempotentRestPublishing, jsonDict, TxClientOptions.idempotentRestPublishing);
+    READ_VALUE(o, fallbackHosts, jsonDict, TxClientOptions.fallbackHosts);
+    READ_VALUE(o, fallbackHostsUseDefault, jsonDict, TxClientOptions.fallbackHostsUseDefault);
+    ON_VALUE(^(const id value) { o.defaultTokenParams = [AblyFlutterReader readTokenParams: value]; }, jsonDict, TxClientOptions.defaultTokenParams);
+    READ_VALUE(o, defaultTokenParams, jsonDict, TxClientOptions.defaultTokenParams);
+    //Following properties not supported by Objective C library
```

Comment:
I think I would prefer that the message conveyed in this comment and the commented out lines below it (even though they would never compile in their current form given the refactor!) was moved somewhere a bit more prominent. Perhaps the developer notes for now? Or an issue - perhaps with an 'investigate' or 'question' label? I like that you want to keep a record of this but it feels lost here.

This also applies to the writer so I've not separately commented there (i.e. `encodeErrorInfo:` method therein).

Reply:
@tiholic ?

(oh, and also, please use proper form "Objective-C", never "Objective C" or - in my opinion worse - "Obj-C" 🤢 )

Reply:
Yes. I wasn't able to add a review comment earlier. This has been resolved.